lib/plugin/selenoid.js

```diff
@@ -0,0 +1,155 @@
+  } = config;
+  const passedTests = [];
+  recorder.startUnlessRunning();
+  replaceScriptConfig({ name, additionalParams });
+
+  if (autoStart) {
+    event.dispatcher.on(event.all.before, () => {
+      recorder.add('Starting selenoid', async () => {
+        return createAndStart(autoCreate).then(() => output.debug('Selenoid started')).catch((err) => { throw err; });
+      });
+    });
+
+    event.dispatcher.on(event.all.after, () => {
+      recorder.add('Stopping selenoid', async () => {
+        return wait(10000).then(() => deletePassedTests(passedTests)).then(stopSelenoid).then(() => output.debug('Selenoid stopped'))
+          .catch((err) => { throw err; });
+      });
+    });
+  }
+
+  event.dispatcher.on(event.all.before, () => {
+    setSelenoidOptions(config);
+  });
+
+  event.dispatcher.on(event.test.before, (test) => {
+    const WebDriver = container.helpers('WebDriver');
+    const { options } = WebDriver;
+    recorder.add('setting selenoid capabilities', () => {
+      options.capabilities['selenoid:options'].name = test.title;
+      options.capabilities['selenoid:options'].videoName = `${test.title}.mp4`;
+      options.capabilities['selenoid:options'].logName = `${test.title}.log`;
+      WebDriver._setConfig(options);
+    });
+  });
+
+  if (deletePassed) {
+    event.dispatcher.on(event.test.passed, (test) => {
+      passedTests.push(test.title);
+    });
+  }
+};
+
+/**
+ * # Selenoid
+ *
+ * Selenoid plugin with for recording video recording and logs
+ *
+ * ## Prerequisite
+ *
+ *  - Docker
+ *
+ * ## How to ?
+ *
+ * ### 1. Add browsers.json
+ * Add browsers.json parallel to codecept conf location. [Refer here](https://aerokube.com/selenoid/latest/#_prepare_configuration) to know more about browsers.json
+ *
+ * **You can download the sample json from example.**
+ *
+ * ### 2. Add plugin configuration in codecept
+ *
+ *  Add plugin config to codecept conf.
+ * ```js
+ * plugins: {
+ *     selenoid: {
+ *       require: '../lib/index',
+ *       enabled: true,
+ *       name: 'testnoid',
+ *       deletePassed: true,
+ *       autoCreate: true,
+ *       autoStart: true,
+ *       sessionTimeout: '30m',
+ *       enableVideo: true,
+ *       enableLog: true,
+ *       additionalParams: '--env TEST=test',
+ *     },
+ *   }
+ * ```
+ * #### Options:
+ * | Param | Description |
+ * |--|--|
+ * | name | Name of the container |
+ * | deletePassed | Delete video and logs of passed tests |
+ * | autoCreate | Will automatically create container (Linux only)|
+ * | autoStart | If disabled start the container manually before running tests |
+ * | enableVideo | Enable video recording (`video` folder of output)|
+ * | enableLog | Enable video recording (`logs` folder of output) |
+ * | additionalParams | [Refer here](https://docs.docker.com/engine/reference/commandline/create/) to know more |
+ *
+ * ### 3. Create selenoid container
+ * **If you are using linux machine, we can handle this for you.**
```

Comment:
Sorry, this line is not clear.

Write about `autoCreate` and this is the simplest way to start. If somehow it doesn't work -> create container manually